pkg/code/data/nonce/tests/tests.go

```diff
@@ -80,6 +84,52 @@ func testUpdate(t *testing.T, s nonce.Store) {
 	assert.EqualValues(t, 1, actual.Id)
 }
 
+func testUpdateInvalid(t *testing.T, s nonce.Store) {
+	ctx := context.Background()
+
+	for _, invalid := range []*nonce.Record{
+		{},
+		{
+			Address: "test_address",
+		},
+		{
+			Address:   "test_address",
+			Authority: "test_authority",
+		},
+		{
+			Address:   "test_address",
+			Authority: "test_authority",
+			Blockhash: "block_hash",
+		},
+		{
+			Address:   "test_address",
+			Authority: "test_authority",
+			Blockhash: "test_blockhash",
+			Purpose:   nonce.PurposeClientTransaction,
+			State:     nonce.StateClaimed,
+		},
+		{
+			Address:     "test_address",
+			Authority:   "test_authority",
+			Blockhash:   "test_blockhash",
+			Purpose:     nonce.PurposeClientTransaction,
+			State:       nonce.StateClaimed,
+			ClaimNodeId: "my-node",
+		},
+		{
+			Address:        "test_address",
+			Authority:      "test_authority",
+			Blockhash:      "test_blockhash",
+			Purpose:        nonce.PurposeClientTransaction,
+			State:          nonce.StateClaimed,
+			ClaimExpiresAt: time.Now().Add(time.Hour),
+		},
+	} {
+		require.Error(t, invalid.Validate())
```

Comment:
Seems like this wasn't really tested or used much before? Will be using more of it in the upcoming functions I imagine